src/diagnostics_component/utils/file_upload_utils/src/blob_storage_helper.cpp

```diff
@@ -0,0 +1,105 @@
+/**
+ * @file blob_storage_helper.cpp
+ * @brief Implements the interface for interacting with Azure Blob Storage
+ *
+ * @copyright Copyright (c) 2021, Microsoft Corp.
+ */
+#include "blob_storage_helper.hpp"
+
+#include <azure_c_shared_utility/string_token.h>
+#include <azure_c_shared_utility/urlencode.h>
+#include <cstring>
+#include <fstream>
+
+/**
+ * @brief Creates the blob storage client using the information in @p blobInfo and then constructs the object
+ * @param blobInfo information related to the blob storage account
+ * @param maxConcurrency the maximum number of threads that can be working at one time for the client connection
+ */
+AzureBlobStorageHelper::AzureBlobStorageHelper(const BlobStorageInfo& blobInfo, const unsigned int maxConcurrency)
```

Comment:
Windows build is failing because maxConcurrency is unused parameter.
Can it be removed since it's not used at all?
Or can use UNREFERENCED_PARAMETER macro if planning to plumb it through to where threads are being spawned. (personal preference is the former so the code just has what's needed today)
```
D:\a\1\s\src\diagnostics_component\utils\file_upload_utils\src\blob_storage_helper.cpp(19,100):
warning C4100: 'maxConcurrency': unreferenced formal parameter 
[D:\a\1\s\out\Debug\src\diagnostics_component\utils\file_upload_utils\file_upload_utility.vcxproj]
```